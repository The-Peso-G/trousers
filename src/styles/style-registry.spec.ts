import StyleRegistry from './style-registry';

describe('StyleRegistry', () => {
    let registry: StyleRegistry;
    let element: HTMLHeadElement;
    let attributeId: string;

    beforeEach(() => {
        element = document.createElement('head');
        attributeId = 'data-testing-trousers';

        registry = new StyleRegistry(element, attributeId);
    });

    it('registers a new style', () => {
        const key = '1';
        const style = 'background-color:red;';

        registry.register(key, style);

        expect(element.innerHTML).toMatchSnapshot();
    });

    it('registers multiple styles', () => {
        registry.register('1', 'background-color:red;');
        registry.register('2', 'background-color:blue;');

        expect(element.innerHTML).toMatchSnapshot();
    });

    it('registering an existing key does not update the style', () => {
        const key = '1';
        const style = 'background-color:red;';
        const styleBlue = 'background-color:blue;';

        registry.register(key, style);
        registry.register(key, styleBlue);

        expect(element.innerHTML).toMatchSnapshot();
    });

    it('detects an existing style', () => {
        const key = '1';
        const style = 'background-color:red;';

        registry.register(key, style);

        expect(registry.has('1')).toBe(true);
    });

    it('detects a non existing style', () => {
        expect(registry.has('123')).toBe(false);
    });
});
