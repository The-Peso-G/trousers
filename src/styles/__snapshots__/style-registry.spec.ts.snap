// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleRegistry registering an existing key does not update the style 1`] = `
<style data-testing-trousers="" type="text/css">
  1{background-color:red;}
</style>
`;

exports[`StyleRegistry registers a new style 1`] = `
<style data-testing-trousers="" type="text/css">
  1{background-color:red;}
</style>
`;

exports[`StyleRegistry registers multiple styles 1`] = `
<style data-testing-trousers="" type="text/css">
  1{background-color:red;}2{background-color:blue;}
</style>
`;
